### ISSUE TITLE
no-async-in-computed-properties should apply to computedAsync too

### ISSUE THREAD

**Issue by nicooprat** [vuejs/eslint-plugin-vue#3019]
**What rule do you want to change?**
no-async-in-computed-properties

**Does this change cause the rule to produce more or fewer warnings?**
more warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**
transparent for users

**Please provide some example code that this change will affect:**

```vue

// fails
const foo = computed(async () => {
  const data = await Promise.resolve('foo');

  return data;
})

// this should fail too
const foo = computedAsync(async () => {
  const data = await Promise.resolve('foo');

  return data;
})
```

**What does the rule currently do for this code?**
nothing

**What will the rule do after it's changed?**
fail

**Additional context**
The VueUse docs states that the same caveat applies: https://vueuse.org/core/computedAsync/#caveats

> Just like Vue's built-in computed function, computedAsync does dependency tracking and is automatically re-evaluated when dependencies change. Note however that only dependencies referenced in the first call stack are considered for this. In other words: Dependencies that are accessed asynchronously will not trigger re-evaluation of the async computed value.

Edit: It's a bit paradoxal because the whole purpose of `computedAsync` is to be async, but it can be hard to understand what's happening when reactivity fails. So I guess this rule can't really be applied, or is meant to be disabled every time once we have "acknowledged" the potential issue...

Feel free to close the issue, at least it can help others if they stumble upon it!

**Comment by ValentinYoushkevich** [comment]
Agree that reporting every `computedAsync` wouldn't help, the callback is async by design. The caveat from the VueUse docs is narrower though: reactive values read after the first `await` aren't tracked. That part could be checked:

```js
const items = computedAsync(async () => {
  const res = await fetch(`/api/items/${id.value}`) // tracked
  const list = await res.json()
  return list.filter((item) => item.type === type.value) // read after await, not tracked
})
```

Would something like that be welcome in this rule, maybe behind an option, or is it out of scope since `computedAsync` is a VueUse API? I can look into it if there's interest.

**Comment by FloEdelmann** [comment]
Hmm I don't know. We don't support VueUse anywhere else and it's not an "official" Vue project (albeit widely used), and in this specific case, the rule would report also for many deliberately correct use cases. I'd slightly prefer closing this as unsupported, what do you say @ota-meshi @waynzh?

**Comment by ValentinYoushkevich** [comment]
Fair enough, fine with closing it.
